streamer/src/quic.rs

```diff
@@ -356,20 +346,23 @@ fn handle_connection(
         while !stream_exit.load(Ordering::Relaxed) {
             match uni_streams.next().await {
                 Some(stream_result) => match stream_result {
-                    Ok(mut stream) => {
+                    Ok(stream) => {
                         stats.total_streams.fetch_add(1, Ordering::Relaxed);
                         stats.total_new_streams.fetch_add(1, Ordering::Relaxed);
-                        let mut maybe_batch = None;
-                        while !stream_exit.load(Ordering::Relaxed) {
-                            if handle_chunk(
-                                &stream.read_chunk(PACKET_DATA_SIZE, false).await,
-                                &mut maybe_batch,
-                                &remote_addr,
-                                &packet_sender,
-                            ) {
-                                last_update.store(timing::timestamp(), Ordering::Relaxed);
-                                break;
+                        if !stream_exit.load(Ordering::Relaxed) {
+                            let timeout_ms = get_read_timeout_ms();
+                            let timeout_read_to_end = timeout(
+                                Duration::from_millis(timeout_ms),
+                                stream.read_to_end(PACKET_DATA_SIZE),
+                            )
+                            .await;
+                            if let Ok(read_to_end) = timeout_read_to_end {
+                                handle_read_to_end(&read_to_end, &remote_addr, &packet_sender);
```

Comment:
```suggestion
                            if let Ok(read_to_end) = timeout_read_to_end.as_ref() {
                                handle_read_to_end(read_to_end, &remote_addr, &packet_sender);
```

to avoid the destructure